### PR TITLE
Small cleanups here and there

### PR DESCRIPTION
**Pull request by Bond-009**
The unrelated changes from #1023 (and more)

### CODE REVIEW

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 1]
```diff
@@ -1841,88 +1832,67 @@ public Task SendMessageToUserSessions<T>(List<Guid> userIds, string name, Func<T
...
                 {
-                    _logger.LogError("Error sending message", ex);
+                    yield return SendMessageToSession(session, name, data, cancellationToken);
                 }
+            }
 
-            }, cancellationToken)).ToArray();
-
-            return Task.WhenAll(tasks);
+            return Task.WhenAll(GetTasks());
```
Comment (JustAMan) [inline Emby.Server.Implementations/Session/SessionManager.cs]: I see this pattern of

1. getting a sessions list
2. defining a `GetTasks` which calls `SendMessageToSession` for all sessions
3. returning `Task.WhenAll(GetTasks())`

repeated at least three times. Can we extract points 2 and 3 in a separate (probably static) method to reduce copy-pasteness?

**Comment by JustAMan** [comment]
@Bond-009 I believe I've made a better title for your PR, please check. Feel free to disagree and revert back if you think old one was better.

File: Emby.Server.Implementations/Session/SessionManager.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Session/SessionManager.cs]: Done 👍

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 2]
```diff
@@ -1037,6 +1037,25 @@ private static async Task SendMessageToSession<T>(SessionInfo session, string na
...
         }
 
+        private static Task SendMessageToSessions<T>(IEnumerable<SessionInfo> sessions, string name, T data, CancellationToken cancellationToken)
+        {
+            IEnumerable<Task> GetTasks()
+            {
+                foreach (var session in sessions)
+                {
+                    var controllers = session.SessionControllers;
+                    var messageId = Guid.NewGuid().ToString("N");
```
Comment (JustAMan) [inline Emby.Server.Implementations/Session/SessionManager.cs]: can we make a `messageId` defined outside of `foreach session` loop?

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 3]
```diff
@@ -1041,11 +1041,10 @@ private static Task SendMessageToSessions<T>(IEnumerable<SessionInfo> sessions,
         {
             IEnumerable<Task> GetTasks()
             {
+                var messageId = Guid.NewGuid().ToString("N");
                 foreach (var session in sessions)
                 {
                     var controllers = session.SessionControllers;
-                    var messageId = Guid.NewGuid().ToString("N");
```
Comment (cvium) [inline Emby.Server.Implementations/Session/SessionManager.cs]: But this changes the behaviour...

Comment (Bond-009) [reply Emby.Server.Implementations/Session/SessionManager.cs]: Correct, but it shouldn't matter.

**Comment by anthonylavado** [comment]
Is this blocked by #1023 then?

**Comment by Bond-009** [comment]
@anthonylavado, no it's the other way around, #1023 is blocked by this.

**Pull request merged**